packages/web-sdk/src/transports/batch/transport.test.ts

```diff
@@ -0,0 +1,109 @@
+import { BaseTransport, LogEvent, LogLevel, TransportItem, TransportItemType } from '@grafana/faro-core';
+import { mockInternalLogger } from '@grafana/faro-core/src/testUtils';
+
+import { BatchTransport } from './transport';
+import type { BatchTransportOptions } from './types';
+
+const item: TransportItem<LogEvent> = {
+  type: TransportItemType.LOG,
+  payload: {
+    context: {},
+    level: LogLevel.INFO,
+    message: 'hi',
+    timestamp: '2023-01-27T09:53:01.035Z',
+  },
+  meta: {},
+} as const;
+
+const fetch = jest.fn(() => Promise.resolve({ status: 200 }));
+(global as any).fetch = fetch;
+
+describe('BatchTransport', () => {
+  afterEach(() => {
+    fetch.mockClear();
+    jest.restoreAllMocks();
+    jest.useRealTimers();
+  });
+
+  it('Sends payload after no new signal has arrived for "batchSendCount" milliseconds after the last one and resets count and payload.', () => {
+    jest.useFakeTimers();
+
+    const mockSendFunction = jest.fn();
+
+    class MockTransport extends BaseTransport {
```

Comment:
nit: Since you are reusing this throughout the tests, you can abstract it into a generator function